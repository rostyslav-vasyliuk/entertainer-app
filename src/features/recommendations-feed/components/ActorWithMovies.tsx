import React from 'react';
import { ScrollView, StyleSheet, View, Text, Image } from 'react-native';
import { TEXT_COLOR, BACKGROUND_LIGHT, TEXT_COLOR_SECONDARY, LIGHT_IMAGE_PLACEHOLDER } from '../../../constants/color-constants';
import { Entypo } from '@expo/vector-icons';
import { movieGenres } from '../../movies/constants';
import { getDateString, getAge } from '../../../constants/date-constants';

const ActorWithMovies = ({ data, navigation }) => {

  const getYear = (releaseDate) => {
    if (!releaseDate) {
      return '';
    }
    return releaseDate.slice(0, 4);
  }

  const getGenre = (genre_id) => {
    if (movieGenres.find((item) => Number(item.movieDB_id) === genre_id)) {
      return movieGenres.find((item) => Number(item.movieDB_id) === genre_id).genre;
    }
  }

  return (
    <View>

      <Text style={{ color: TEXT_COLOR_SECONDARY, paddingBottom: 6, paddingTop: 10, paddingLeft: '4%', textTransform: 'uppercase', fontSize: 16, fontWeight: '600', letterSpacing: 1 }}>
        {'Actor you may like'}
      </Text>

      <View style={actorMoviesStyles.wrapper}>
        <View style={{ flexDirection: 'row', paddingBottom: 12, padding: 10 }}>
          <Image
            source={{ uri: `https://image.tmdb.org/t/p/w1280/${data.actor.profile_path}` }}
            style={{ width: 100, height: 140, borderRadius: 16 }}
          />
          <View style={{ padding: 12, paddingTop: 2, paddingBottom: 2, justifyContent: 'space-between' }}>
            <View>
              <Text style={{ color: TEXT_COLOR, paddingTop: 3, paddingBottom: 4, fontWeight: '600', fontSize: 16, letterSpacing: 1.1 }}>
                {data.actor.name}
              </Text>

              <Text style={{ color: TEXT_COLOR, paddingTop: 3, fontWeight: '400', fontSize: 12, overflow: 'hidden' }}>
                {data.actor.place_of_birth}
              </Text>

              <Text style={{ color: TEXT_COLOR, paddingTop: 3, fontWeight: '400', fontSize: 12, overflow: 'hidden' }}>
                {`Born: ${getDateString(data.actor.birthday)} (${getAge(data.actor.birthday)} y.o.)`}
              </Text>

              <Text style={{ color: TEXT_COLOR, paddingTop: 3, fontWeight: '400', fontSize: 12, overflow: 'hidden' }}>
                {`Movies: ${data.actor.credits.cast.length}`}
              </Text>
            </View>

            {/* <TouchableOpacity>
              <View style={{ justifyContent: 'center', alignItems: 'center', flexDirection: 'row', width: '100%' }}>
                <Text style={{ color: TEXT_COLOR_SECONDARY, fontSize: 12, fontWeight: '600', letterSpacing: 0.9 }}>
                  {'Know more'}
                </Text>

                <Entypo name='chevron-small-right' />
              </View>
            </TouchableOpacity> */}
          </View>
        </View>

        <Text style={{ color: TEXT_COLOR, fontSize: 14, marginLeft: 10, fontWeight: '600', letterSpacing: 0.9 }}>
          {'Best known for:'}
        </Text>
        <ScrollView horizontal showsHorizontalScrollIndicator={false}>
          {data.data.map((element, index) => (
            index < 10 &&
            (
              <View style={{ minHeight: 180, width: 120, marginLeft: 8, marginRight: 4, marginTop: 5, borderRadius: 8, marginBottom: 6 }}>
                {element.backdrop_path ? (
                  <Image
                    source={{ uri: `https://image.tmdb.org/t/p/w1280/${element.backdrop_path}` }}
                    style={{ height: 150, width: 120, borderRadius: 8 }}
                  />
                ) : (
                    <View style={{ height: 150, width: 120, borderRadius: 8, backgroundColor: LIGHT_IMAGE_PLACEHOLDER, justifyContent: 'center', alignItems: 'center' }}>
                      <Entypo name='image' style={{ color: TEXT_COLOR_SECONDARY, fontSize: 36 }} />
                    </View>
                  )
                }

                <Text style={{ color: TEXT_COLOR, fontSize: 12, marginTop: 3, fontWeight: '400' }}>
                  {element.title}
                </Text>

                <Text style={{ color: TEXT_COLOR, fontSize: 11, marginTop: 3, fontWeight: '300' }}>
                  {`${getYear(element.release_date)}, ${getGenre(element.genre_ids[0])}`}
                </Text>
              </View>
            )
          ))}
        </ScrollView>
      </View>
    </View>
  );
}

export default ActorWithMovies;

const actorMoviesStyles = StyleSheet.create({
  wrapper: {
    // padding: 8,
    // paddingRight: 2,
    borderRadius: 16,
    width: '94%',
    backgroundColor: BACKGROUND_LIGHT,
    marginLeft: '3%',
    marginRight: '3%'
  }
});
