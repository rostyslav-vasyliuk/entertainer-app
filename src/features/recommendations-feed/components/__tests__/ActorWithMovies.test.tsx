import React from 'react';
import ActorWithMovies from '../ActorWithMovies';
import renderer from 'react-test-renderer';

describe('<AppIntroOverlay />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<ActorWithMovies navigation={{}} data={{ data: [], actor: { credits: { cast: [] } } }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has 1 child', () => {
    const tree = renderer.create(<ActorWithMovies navigation={{ navigate: () => { } }} data={{ data: [], actor: { credits: { cast: [] } } }} />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it('has 1 child', () => {
    const tree = renderer.create(<ActorWithMovies navigation={{ navigate: () => { } }} data={{ data: [], actor: { credits: { cast: [] } } }} />).toJSON();
    expect(tree.children.length).toBeGreaterThanOrEqual(2);
  });
});
