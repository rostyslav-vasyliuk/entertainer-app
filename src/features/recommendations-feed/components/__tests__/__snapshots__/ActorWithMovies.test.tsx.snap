// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppIntroOverlay /> match snapshot 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#c0c1c2",
        "fontSize": 16,
        "fontWeight": "600",
        "letterSpacing": 1,
        "paddingBottom": 6,
        "paddingLeft": "4%",
        "paddingTop": 10,
        "textTransform": "uppercase",
      }
    }
  >
    Actor you may like
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#181821",
        "borderRadius": 16,
        "marginLeft": "3%",
        "marginRight": "3%",
        "width": "94%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 10,
          "paddingBottom": 12,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w1280/undefined",
          }
        }
        style={
          Object {
            "borderRadius": 16,
            "height": 140,
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "padding": 12,
            "paddingBottom": 2,
            "paddingTop": 2,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
                "letterSpacing": 1.1,
                "paddingBottom": 4,
                "paddingTop": 3,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
                "fontWeight": "400",
                "overflow": "hidden",
                "paddingTop": 3,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
                "fontWeight": "400",
                "overflow": "hidden",
                "paddingTop": 3,
              }
            }
          >
            Born:  ( y.o.)
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
                "fontWeight": "400",
                "overflow": "hidden",
                "paddingTop": 3,
              }
            }
          >
            Movies: 0
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "600",
          "letterSpacing": 0.9,
          "marginLeft": 10,
        }
      }
    >
      Best known for:
    </Text>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`<AppIntroOverlay /> match snapshot 2`] = `
<View>
  <Text
    style={
      Object {
        "color": "#c0c1c2",
        "fontSize": 16,
        "fontWeight": "600",
        "letterSpacing": 1,
        "paddingBottom": 6,
        "paddingLeft": "4%",
        "paddingTop": 10,
        "textTransform": "uppercase",
      }
    }
  >
    Actor you may like
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#181821",
        "borderRadius": 16,
        "marginLeft": "3%",
        "marginRight": "3%",
        "width": "94%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 10,
          "paddingBottom": 12,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w1280/undefined",
          }
        }
        style={
          Object {
            "borderRadius": 16,
            "height": 140,
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "padding": 12,
            "paddingBottom": 2,
            "paddingTop": 2,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
                "letterSpacing": 1.1,
                "paddingBottom": 4,
                "paddingTop": 3,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
                "fontWeight": "400",
                "overflow": "hidden",
                "paddingTop": 3,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
                "fontWeight": "400",
                "overflow": "hidden",
                "paddingTop": 3,
              }
            }
          >
            Born:  ( y.o.)
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
                "fontWeight": "400",
                "overflow": "hidden",
                "paddingTop": 3,
              }
            }
          >
            Movies: 0
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "600",
          "letterSpacing": 0.9,
          "marginLeft": 10,
        }
      }
    >
      Best known for:
    </Text>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
